### PR TITLE
Update SIP protocol

### PR DESCRIPTION
**Pull request by dennwc**
Update SIP protocol:
- Pass custom headers
- Allow mapping response headers to participant attributes
- Pass project ID for Cloud
- Pass FQDN for outbound

### CODE REVIEW

File: protobufs/rpc/io.proto [hunk 1]
```diff
@@ -151,7 +155,15 @@ message EvaluateSIPDispatchRulesResponse {
   string sip_trunk_id = 9;
   // Dispatch Rule used to fulfill this request
   string sip_dispatch_rule_id = 10;
-  // NEXT ID: 12
+
+  // Used in Cloud only
+  string project_id = 12;
+
+  // Include these SIP X-* headers in 200 OK response.
+  map<string, string> headers = 13;
```
[the remark this answers is not in the document]
Comment (davidzhao) [reply protobufs/rpc/io.proto:163]: if it's response, it'd be great to clarify them. same as below.

[the remark this answers is not in the document]
Comment (dennwc) [reply protobufs/rpc/io.proto:163]: A few users from the community asked about adding specific headers to our responses. I think it helps them handle calls to/from LiveKit in a special way. 

Correct, they specify headers directly here and we pass them unchanged to the other SIP endpoint. These headers are set on the first message that we send. For inbound, it's a response to INVITE (200 OK), and for outbound it's the INVITE request itself.

I'll add more docs to these 👍

File: protobufs/livekit_sip.proto [hunk 2]
```diff
@@ -206,6 +213,27 @@ message SIPOutboundTrunkInfo {
...
   string auth_username = 7;
   string auth_password = 8;
+
+  // Include these SIP X-* headers in INVITE request.
+  map<string, string> headers = 9;
+  // Map SIP X-* headers from 200 OK to SIP participant attributes.
+  map<string, string> headers_to_attributes = 10;
+}
+
+message GetSIPInboundTrunkRequest {
```
Comment (biglittlebigben) [inline protobufs/livekit_sip.proto:223]: It's unclear to me what TrunkRequests and TrunkResponses are?

Comment (dennwc) [reply protobufs/livekit_sip.proto:223]: You mean it's unclear why they are here? I just noticed that our OSS API for managing Trunks supports listing trunks, but cannot get a Trunk by ID. So I'm adding it here.

Comment (biglittlebigben) [reply protobufs/livekit_sip.proto:223]: Never mind, I misread here.

**Review by biglittlebigben — APPROVED** [review]

File: protobufs/rpc/io.proto (hunk 1 above)
[the remark this answers is not in the document]
Comment (dennwc) [reply protobufs/rpc/io.proto:164]: Done 👍

**Review by davidzhao — APPROVED** [review]

**Pull request merged**